mmv1/third_party/validator/kms_key_ring_iam.go

```diff
@@ -0,0 +1,71 @@
+package google
+
+import "fmt"
+
+func GetKmsKeyRingIamPolicyCaiObject(d TerraformResourceData, config *Config) ([]Asset, error) {
+	return newKmsKeyRingIamAsset(d, config, expandIamPolicyBindings)
+}
+
+func GetKmsKeyRingIamBindingCaiObject(d TerraformResourceData, config *Config) ([]Asset, error) {
+	return newKmsKeyRingIamAsset(d, config, expandIamRoleBindings)
+}
+
+func GetKmsKeyRingIamMemberCaiObject(d TerraformResourceData, config *Config) ([]Asset, error) {
+	return newKmsKeyRingIamAsset(d, config, expandIamMemberBindings)
+}
+
+func MergeKmsKeyRingIamPolicy(existing, incoming Asset) Asset {
+	existing.IAMPolicy = incoming.IAMPolicy
+	return existing
+}
+
+func MergeKmsKeyRingIamBinding(existing, incoming Asset) Asset {
+	return mergeIamAssets(existing, incoming, mergeAuthoritativeBindings)
+}
+
+func MergeKmsKeyRingIamBindingDelete(existing, incoming Asset) Asset {
+	return mergeDeleteIamAssets(existing, incoming, mergeDeleteAuthoritativeBindings)
+}
+
+func MergeKmsKeyRingIamMember(existing, incoming Asset) Asset {
+	return mergeIamAssets(existing, incoming, mergeAdditiveBindings)
+}
+
+func MergeKmsKeyRingIamMemberDelete(existing, incoming Asset) Asset {
+	return mergeDeleteIamAssets(existing, incoming, mergeDeleteAdditiveBindings)
+}
+
+func newKmsKeyRingIamAsset(
+	d TerraformResourceData,
+	config *Config,
+	expandBindings func(d TerraformResourceData) ([]IAMBinding, error),
+) ([]Asset, error) {
+	bindings, err := expandBindings(d)
+	if err != nil {
+		return []Asset{}, fmt.Errorf("expanding bindings: %v", err)
+	}
+
+	name, err := assetName(d, config, "//cloudkms.googleapis.com/{{key_ring_id}}")
```

Comment:
I think this is probably correct, but it might be good to double-check while verifying the expected asset name for crypto keys.

Reply:
cloudkms.googleapis.com is the correct one. it is verified already.. kms.googleapis.com is incorrect and an existing bug that i have a fix in another MR.

Reply:
@palani-ram-google-partner yes, `cloudkms.googleapis.com` is correct; the question is whether there should be anything after the domain besides `key_ring_id`. I _think_ this is probably correct as-is but if you've done any investigation to be sure that would be great to know.

Reply:
key_ring_id is enough, it works perfectly. i have a policy developed and tested

Reply:
You're correct, `key_ring_id` is the only option here: resources/google_kms_key_ring_iam#key_ring_id